pygen/pygen_src/isa/riscv_instr.py

```diff
@@ -166,14 +168,66 @@ def create_csr_filter(self, cfg):
             else:                                              # User Mode
                 self.include_reg.append("USCRATCH")
 
-    def get_rand_instr(self):
-        pass  # TODO
+    def get_rand_instr(self, include_instr=[], exclude_instr=[],
+                       include_category=[], exclude_category=[],
+                       include_group=[], exclude_group=[]):
+        idx = BitArray(uint = 0, length = 32)
+        name = ""
+        allowed_instr = []
+        disallowed_instr = []
+        # allowed_categories = []
+
+        for items in include_category:
+            allowed_instr.append(self.instr_category[items])
+
+        for items in exclude_category:
+            if(items in self.instr_category):
+                disallowed_instr.append(self.instr_category[items])
+        for items in include_group:
+            allowed_instr.append(self.instr_group[items])
+        for items in exclude_group:
+            if(items in self.instr_group):
+                disallowed_instr.append(self.instr_group[items])
+
+        disallowed_instr.extend(exclude_instr)
+
+        if(len(disallowed_instr) == 0):
+            if(len(include_instr) > 0):
+                idx = random.randrange(0, len(include_instr) - 1)
+                name = include_instr[idx]
+            elif(len(allowed_instr > 0)):
+                idx = random.randrange(0, len(allowed_instr) - 1)
+                name = allowed_instr[idx]
+            else:
+                idx = random.randrange(0, len(self.instr_names) - 1)
+                name = self.instr_names[idx]
+        else:
+            instr_names_set = set(self.instr_names)
+            disallowed_instr_set = set(disallowed_instr)
+            allowed_instr_set = set(allowed_instr)
+            include_instr_set = set(include_instr)
+            excluded_instr_names_list = list(instr_names_set - disallowed_instr_set)
+            excluded_allowed_instr_list = list(allowed_instr_set - disallowed_instr_set)
+            include_instr_list = list(include_instr_set - disallowed_instr_set)
+
+            name = random.choice(excluded_instr_names_list)
```

Comment:
I think the approach you use here is ok temporarily, but we will need a more robust system to handle the more complex constraints used elsewhere in the generator.
Can you add a TODO here to switch this randomization method over to whichever solution we decide to go with later on?

Reply:
Sure.